src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,47 @@
+import { Todo } from '../../types/Todo';
+import { TodoItem } from '../TodoItem/TodoItem';
+
+type Props = {
+  todos: Todo[];
+  removeTodo: (id: number) => Promise<void>;
+  isAdding: boolean;
+  isDeleting: boolean;
+};
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  removeTodo,
+  isAdding,
+  isDeleting,
+}) => {
+  const temporaryTodo = {
```

Comment:
why do you create `temporaryTodo ` in this component if you need it in `TodoItem`?
in the Props type use `?` for no required prop.
and use the ternary operator to check if you pass or not todo and use your tempTodo instead undefined

Reply:
Because i wanna add temporary todo when i add new todo and waiting for response. I have no idea how i can realize that you told me in TodoItem

Reply:
I can't find the code where you added `temporaryTodo` to the end of todos array.

In this situation, I suggest, better:
- create `temporaryTodo` in `addTodo`(App.tsx)
- set it to the end of todos array;
- use `finally` to filter todos(no todo with id=0 should left)

And do not forget id= 0, not -1.
<img width="838" alt="Screenshot 2022-10-11 at 00 17 04" src="https://user-images.githubusercontent.com/17098736/194956958-3de0238e-e714-4a3e-ac8c-a8d53252bbe0.png">

Be attentive with task descriptions. There **no redundant** words. The same would be in test tasks. You need to read in attentively and write the code according to it. It is important to pass all employment stages and get the job 😉